src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -185,6 +189,32 @@ private StatementSyntax Transform( StatementSyntax stmt )
 			_ => UnhandledSyntax( stmt )
 		};
 
+	private StatementSyntax Transform( ReturnStatementSyntax returnStmt ) {
+		var expr = returnStmt.Expression;
+		if( m_generatedFunctionReturnsVoid && expr is not null ) {
+			if( IsStatementCompatibleExpression( expr ) ) {
+				return SyntaxFactory.Block(
+						SyntaxFactory.ExpressionStatement( Transform( expr ) ).WithLeadingTrivia( SyntaxFactory.Space ),
+						SyntaxFactory.ReturnStatement().WithLeadingTrivia( SyntaxFactory.Space ).WithTrailingTrivia( SyntaxFactory.Space )
+					)
+					.WithTriviaFrom( returnStmt );
+			} else {
+				return SyntaxFactory.ReturnStatement().WithTriviaFrom( returnStmt );
+			}
+		}
+		return returnStmt.WithExpression( MaybeTransform( returnStmt.Expression, Transform ) );

```

Comment:
An alternative way to write this is:

```suggestion
		if( !m_generatedFunctionReturnsVoid || expr is null ) {
			return returnStmt.WithExpression( MaybeTransform( returnStmt.Expression, Transform ) );
		}
		
		if( IsStatementCompatibleExpression( expr ) ) {
			return SyntaxFactory.Block(
					SyntaxFactory.ExpressionStatement( Transform( expr ) ).WithLeadingTrivia( SyntaxFactory.Space ),
					SyntaxFactory.ReturnStatement().WithLeadingTrivia( SyntaxFactory.Space ).WithTrailingTrivia( SyntaxFactory.Space )
			).WithTriviaFrom( returnStmt );
		}
		
		return SyntaxFactory.ReturnStatement().WithTriviaFrom( returnStmt );
```

i.e. by flipping the condition in the first `if` and returning we can take the other case and rather than having an else just flatten one level of indentation. And then that other final else also becomes optional because the inside of the `if` always returns.

The benefit of this style is (1) less indentation overall, which can make it easier to read (2) smaller stack trace in your head that you have to unwind, like as you nest ifs you need to push/pop assumptions about the variables you're checking in the condition, whereas in the flattened one you are only ever accumulating facts (not popping) as you go along, like the context becomes more and more specific the longer the function goes which is in a way simpler (imagine playing "20 questions", after a bunch of questions as long as you can look at the previous questions you've asked (look at the preceeding if statements), the fewer possible worlds you are in as you've asked more questions.)

Just a style thing though/matter of taste

Reply:
If you google "return early" you can get an infinite amount of opinions on this